develop/items/first-item.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Creating Your First Item
+description: Learn how to register a simple item and how to texture, model and name it.
+authors:
+  - IMB11
+---
+
+# Creating Your First Item
+
+This page will introduce you into some key concepts relating to items, and how you can register, texture, model and name them.
+
+If you aren't aware, everything in Minecraft is stored in registries, and items are no exception to that.
+
+## Preparing Your Items Class
+
+To simplify the registering of items, you can create a method that accepts an instance of an item and a string identifier.
+
+You can put this method in a class called `ModItems` (or whatever you want to name the class!). 
+
+Mojang does this with their items as well! Check out the `Items` class for inspiration.
+
+```java
+public class ModItems {
+    // We can use generics to make it so we dont need to 
+    // cast to an item when using this method.
+    public static <T extends Item> T register(T item, String ID) {
+        // Create the identifier for the item.
+        Identifier itemID = new Identifier("mod_id", ID);
+        
+        // Register the item.
+        T registeredItem = Registry.register(Registries.ITEM, itemID, item);
+
+        // Return the registered item!
+        return registeredItem;
+    }
+}
+```
+
+## Registering an Item
+
+You can now register an item using the method now.
+
+The item constructor takes in an instance of the `Items.Settings` class as a parameter. This class allows you to:
+
+- Assign the item's `ItemGroup`.
+- Make the item edible by passing a `FoodComponent`.
+- Set the item's durability.
+- ...and other miscellaneous properties.
+
+@[code transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/item/ModItems.java)
+
+However, when you go in-game, you can see that our item doesn't exist! This is because you don't statically initialize the class.
+
+To do this, you can add a public static initialize method to your class and call it from your `ModInitializer` class. Currently, this method doesn't need anything inside it.
+
+@[code transcludeWith=:::3](@/reference/latest/src/main/java/com/example/docs/item/ModItems.java)
+
+@[code transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/item/FabricDocsReferenceItems.java)
+
+Calling a method on a class statically initializes it if it hasn't been used before - this means that all `final` fields are evaluated.
```

Comment:
Aren't all `static` fields initialized there? Technically they are also final here, but they are not initialized because they are final, but they are static, no?

- On a class level, `final` blocks extending from that class.
- On a method level, `final` blocks overriding in subclasses
- On a variable level, `final` blocks the reassigning to other values

Reply:
Yep, will edit tomorrow.

Reply:
You probably know that already, but for newer devs, I usually explain it like this:

Static means that the field doesn't belong to any object, but rather to the class itself.

To be loaded, a non-static field would need their object to be loaded, while a static field *(and also the static block in a class)* need only their class to be accessed, to be loaded in. That's why the `initialize` method can stay empty. It's only there to load the class. 

Some like to put their logger statements in there to see if / when stuff gets loaded